docs/spec/errors/v2_error.json

```diff
@@ -13,22 +13,30 @@
                     "maxLength": 1024
                 },
                 "trace_id": {
-                    "description": "Hex encoded 128 random bits ID of the correlated trace.", 
+                    "description": "Hex encoded 128 random bits ID of the correlated trace. Must be present if transaction_id and parent_id are set.", 
                     "type": ["string", "null"],
                     "maxLength": 1024
                 },
                 "transaction_id": {
```

Comment:
Do we really need the `transaction_id` here? If possible, I'd like to avoid that. We currently use the `transaction_id` on spans to make it easier for the UI to display a transaction view. I don't think the errors also need a `transaction_id`. The problem with adding the `transaction_id` is that it makes the case really hard where a span is started after the transaction has ended and makes in-process context propagation more awkward.

@axw might also have a opinion about that

Reply:
I'm currently experimenting with adding a `transactionId` field to the [`TraceContext`](https://github.com/elastic/apm-agent-java/blob/master/apm-agent-core/src/main/java/co/elastic/apm/impl/transaction/TraceContext.java) class. That would ease the handling quite a bit.

Reply:
Having the `transaction_id` might be useful if one wants to show all errors for a transaction. Do we plan this @sqren ?
I am ok with making it optional, I am relying on the knowledge of the @elastic/apm-agent-devs regarding which attributes can be provided and which not.

Reply:
> Having the transaction_id might be useful if one wants to show all errors for a transaction

Yes, we have talked linking from errors -> transactions, and vice versa.  Eg. it would probably be useful to show the error in the timeline somehow. If the error does not have a `transaction_id` I'm not sure how we can easily query for it.

Btw. Will `transaction_id` -> `transaction.id`, and `trace_id` -> `trace.id` to align with https://github.com/elastic/apm-server/pull/1393?

Reply:
Turned out adding the transaction id to TraceContext works quite well: https://github.com/elastic/apm-agent-java/pull/217

Reply:
Neat!